doc/build/snippets/writing.rst

```diff
@@ -0,0 +1,162 @@
+this page.
+
+You can combine settings whenever they appear under the same keys. For example,
+you can combine a snippet-specific devicetree overlay and Kconfig fragment like
+this:
+
+.. code-block:: yaml
+
+   name: foo
+   append:
+     DTC_OVERLAY_FILE: foo.overlay
+     OVERLAY_CONFIG: foo.conf
+
+Where snippets are located
+**************************
+
+The build system looks for snippets in these places:
+
+#. In directories configured by the :makevar:`SNIPPET_ROOT` CMake variable.
+   This always includes the zephyr repository (so
+   :zephyr_file:`zephyr/snippets` is always a source of snippets) and the
+   application source directory (so :file:`<app>/snippets` is also).
+
+   Additional directories can be added manually at CMake time.
+
+   The variable is a whitespace- or semicolon-separated list of directories
+   which may contain snippet definitions.
+
+   For each directory in the list, the build system looks for
+   :file:`snippet.yml` files underneath a subdirectory named :file:`snippets/`,
+   if one exists.
+
+   For example, if :makevar:`SNIPPET_ROOT` is set to ``/foo;/bar``, the build
+   system will look for :file:`snippet.yml` files underneath the following
+   subdirectories:
+
+   - :file:`/foo/snippets/`
+   - :file:`/bar/snippets/`
+
+   The :file:`snippet.yml` files can be nested anywhere underneath these
+   locations.
+
+#. In any :ref:`module <modules>` whose :file:`module.yml` file provides a
+   ``snippet_root`` setting.
+
+   For example, in a zephyr module named ``baz``, you can add this to your
+   :file:`module.yml` file:
+
+   .. code-block:: yaml
+
+      settings:
+        snippet_root: .
+
+   And then any :file:`snippet.yml` files in ``baz/snippets`` will
+   automatically be discovered by the build system, just as if
+   the path to ``baz`` had appeared in :makevar:`SNIPPET_ROOT`.
+
+Processing order
+****************
+
+The order that snippets are processed is currently not defined.
+Therefore, you should write your :file:`snippet.yml` file so that
+it is not dependent on other snippets.
+
+.. _snippets-devicetree-overlays:
+
+Devicetree overlays (``.overlay``)
+**********************************
+
+This :file:`snippet.yml` adds :file:`foo.overlay` to the build:
+
+.. code-block:: yaml
+
+   name: foo
+   append:
+     DTC_OVERLAY_FILE: foo.overlay
+
+The path to :file:`foo.overlay` is relative to the directory containing
+:file:`snippet.yml`.
+
+.. _snippets-conf-files:
+
+Kconfig fragments (``.conf``)
+*****************************
+
+This :file:`snippet.yml` adds :file:`foo.conf` to the build:
+
+.. code-block:: yaml
+
+   name: foo
+   append:
+     OVERLAY_CONFIG: foo.conf
+
+The path to :file:`foo.conf` is relative to the directory containing
+:file:`snippet.yml`.
+
+Board-specific settings
+***********************
+
+You can write settings that only apply to some boards.
+
+By name
+=======
+
+.. code-block:: yaml
+
+   name: ...
+   boards:
+     bar: # settings for board "bar" go here
```

Comment:
A board name seems to work fine, but a revision does not:
```
name: cdc-acm-console
boards:
  nrf9160dk_nrf9160@0.7.0:
    append:
      OVERLAY_CONFIG: cdc-acm-console.conf
      DTC_OVERLAY_FILE: cdc-acm-console.overlay
```
Does not seem to be picked up when configuring with `-DBOARD=nrf9160dk_nrf9160@0.7.0` (build should fail)

Reply:
Yeah, I wanted to get your and @tejlmand 's opinion on how board revisions should be handled.

I lean towards a syntax that looks like this instead:

```
name: foo
boards:
  bar:
    revision: 1.2.3
      DTC_OVERLAY_FILE: bar-1.2.3.overlay
```

The `@` syntax is OK for a `BOARD` shorthand, but I think something more explicit would be easier to read in snippet.yml.

Reply:
That looks like a good way to have it.